### PR TITLE
update subjects module

### LINKED ISSUE #64: update Subjects modules

### PR DESCRIPTION
**Pull request by VoitV**

### CODE REVIEW

**Review by natamrshn — APPROVED** [review]

**Review by nadraharr — APPROVED** [review]

File: docs/subjects.yaml [hunk 1]
```diff
@@ -0,0 +1,297 @@
+paths:
+  /subject:
```
Comment (vlad-khrychov) [inline docs/subjects.yaml]:
```suggestion
  /subjects:
```

File: docs/subjects.yaml [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+                        category:
+                          type: string
+                          description: The category id the subject belongs to
+                          example: "670fbe3b702a8fef395d6d6c"
+        '401':
+          description: Unauthorized, missing or invalid token
+        '500':
+          description: Internal server error
+
+  /subject/{id}:
```
Comment (vlad-khrychov) [inline docs/subjects.yaml]:
```suggestion
  /subjects/{id}:
```

File: docs/subjects.yaml [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+                    description: The category id the subject belongs to
+                    example: "670fbe3b702a8fef395d6d6c"
+        '404':
+          description: Subject not found
+        '401':
+          description: Unauthorized, missing or invalid token
+        '500':
+          description: Internal server error
+
+  /subject/categories/{categoryId}/subjects:
```
Comment (vlad-khrychov) [inline docs/subjects.yaml]:
```suggestion
  /categories/{categoryId}/subjects:
```

File: docs/subjects.yaml [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+        '400':
+          description: Bad request, invalid or missing category ID
+        '401':
+          description: Unauthorized, missing or invalid token
+        '404':
+          description: No subjects found for the given category ID
+        '500':
+          description: Internal server error
+
+  /subject:
```
Comment (vlad-khrychov) [inline docs/subjects.yaml]:
```suggestion
  /subjects:
```

File: docs/subjects.yaml [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+                    description: The category ID the subject belongs to
+                    example: "670fbe3b702a8fef395d6d6c"
+        '400':
+          description: Bad request, invalid data
+        '401':
+          description: Unauthorized, missing or invalid token
+        '500':
+          description: Internal server error
+
+  /subject/{id}:
```
Comment (vlad-khrychov) [inline docs/subjects.yaml]:
```suggestion
  /subjects/{id}:
```

File: src/modules/category/category.service.js [hunk 6]
```diff
@@ -21,7 +22,17 @@ const categoryService = {
   },
   findOneById: async (id) => {
     return await Category.findById(id).lean().exec()
-  }
+  },
+  getSubjectsByCategoryId: async (categoryId, sort, skip = 0, limit = 10) => {
```
Comment (vlad-khrychov) [inline src/modules/category/category.service.js:26]: What about default value for `sort`?

**Review by vlad-khrychov — APPROVED** [review]

**Review by zim89 — APPROVED** [review]

**Pull request merged**